src/main/java/teammates/storage/sqlentity/User.java

```diff
@@ -0,0 +1,149 @@
+package teammates.storage.sqlentity;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.Objects;
+
+import org.hibernate.annotations.CreationTimestamp;
+import org.hibernate.annotations.UpdateTimestamp;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Inheritance;
+import jakarta.persistence.InheritanceType;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.OneToOne;
+import jakarta.persistence.Table;
+
+@Entity
+@Table(name = "Users")
+@Inheritance(strategy = InheritanceType.JOINED)
+public abstract class User extends BaseEntity {
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private int id;
+
+    @OneToOne
+    @JoinColumn(name = "accountId")
+    private Account account;
+
+    @OneToOne
+    @JoinColumn(name = "courseId")
+    private Course course;
```

Comment:
These should not OneToOne either. OneToOne means one student is associated to one course and one course is associated with one student, which is not what we want here.

Reply:
@samuelfangjw Thanks for catching this!

Have changed Course to OneToMany. Please correct me if im wrong! I read it as One User is associated to Many Courses.

How should we read it for Account and User? I thought One User is associated to One Account, unless im missing something...!

Reply:
One account can be associated with multiple users

Reply:
@samuelfangjw Thanks! I fixed the associations, hope I'm interpreting it correctly!